[Bug] startup failure caused by Retryer

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.


### Environment

Windows

### EventMesh version

master

### What happened

At commit https://github.com/apache/eventmesh/commit/109c493776927e2a50549b7162f95ec8cdca4031, startup success.

At commit https://github.com/apache/eventmesh/commit/08fabd20ec6fcf23ad8197eb21b9f9ffec96e3ea (https://github.com/apache/eventmesh/issues/4060), startup failure.

@pandaapo May you please help resolve this issue?

### How to reproduce

master branch, no customized config.

### Debug logs

2023-09-24 23:34:53,692 INFO  [main] EventMeshGrpcServer(EventMeshGrpcServer.java:133) - GRPCServer[port=10205] started
2023-09-24 23:34:53,692 INFO  [main] EventMeshGrpcServer(EventMeshGrpcServer.java:134) - -----------------EventMeshGRPCServer initialized
2023-09-24 23:34:53,693 INFO  [main] EventMeshServer(EventMeshServer.java:142) - eventStore : null
2023-09-24 23:34:53,694 INFO  [main] ProducerTopicManager(ProducerTopicManager.java:52) - ProducerTopicManager inited......
2023-09-24 23:34:53,694 INFO  [main] EventMeshServer(EventMeshServer.java:149) - server state:INITED
2023-09-24 23:34:53,709 INFO  [main] HTTPMetricsServer(HTTPMetricsServer.java:70) - Register httpMetrics to org.apache.eventmesh.metrics.prometheus.PrometheusMetricsRegistry
2023-09-24 23:34:53,710 INFO  [main] HTTPMetricsServer(HTTPMetricsServer.java:94) - HTTPMetricsServer started.
2023-09-24 23:34:53,710 INFO  [main] ConsumerManager(ConsumerManager.java:56) - consumerManager started......
2023-09-24 23:34:53,710 INFO  [main] ProducerManager(ProducerManager.java:44) - producerManager started......
2023-09-24 23:34:53,716 pool-1-thread-1 DEBUG Stopping LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@55634720]
2023-09-24 23:34:53,716 pool-1-thread-1 DEBUG Stopping LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@55634720]...
2023-09-24 23:34:53,710 ERROR [main] EventMeshStartup(EventMeshStartup.java:57) - EventMesh start fail.
java.lang.NullPointerException: null
	at org.apache.eventmesh.runtime.core.protocol.AbstractRetryer.start(AbstractRetryer.java:77) ~[eventmesh-runtime-1.9.0-release.jar:1.9.0-release]
	at org.apache.eventmesh.runtime.boot.EventMeshHTTPServer.start(EventMeshHTTPServer.java:163) ~[eventmesh-runtime-1.9.0-release.jar:1.9.0-release]
	at org.apache.eventmesh.runtime.boot.EventMeshHttpBootstrap.start(EventMeshHttpBootstrap.java:54) ~[eventmesh-runtime-1.9.0-release.jar:1.9.0-release]
	at org.apache.eventmesh.runtime.boot.EventMeshServer.start(EventMeshServer.java:165) ~[eventmesh-runtime-1.9.0-release.jar:1.9.0-release]
	at org.apache.eventmesh.runtime.boot.EventMeshStartup.main(EventMeshStartup.java:40) [eventmesh-runtime-1.9.0-release.jar:1.9.0-release]
	at org.apache.eventmesh.starter.StartUp.main(StartUp.java:25) [main/:?]
2023-09-24 23:34:53,719 pool-1-thread-1 DEBUG Shutting down OutputStreamManager SYSTEM_OUT.false.false
2023-09-24 23:34:53,720 pool-1-thread-1 DEBUG OutputStream closed
2023-09-24 23:34:53,720 pool-1-thread-1 DEBUG Shut down OutputStreamManager SYSTEM_OUT.false.false, all resources released: true
2023-09-24 23:34:53,720 pool-1-thread-1 DEBUG Appender console stopped with status true
2023-09-24 23:34:53,720 pool-1-thread-1 DEBUG Log4j2 ConfigurationScheduler shutting down threads in java.util.concurrent.ScheduledThreadPoolExecutor@5ef3405a[Running, pool size = 1, active threads = 0, queued tasks = 1, completed tasks = 0]
2023-09-24 23:34:53,721 pool-1-thread-1 DEBUG Stopped XmlConfiguration[location=D:\Dev\eventmesh\eventmesh-runtime\conf\log4j2.xml] OK
2023-09-24 23:34:53,721 pool-1-thread-1 DEBUG Stopped LoggerContext[name=73d16e93, org.apache.logging.log4j.core.LoggerContext@55634720] with status true

> Task :eventmesh-starter:StartUp.main() FAILED

Execution failed for task ':eventmesh-starter:StartUp.main()'.
> Process 'command 'C:\Program Files\Java\jdk-1.8\bin\java.exe'' finished with non-zero exit value -1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
BUILD FAILED in 1m 26s
68 actionable tasks: 68 executed

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

--- a/eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/AbstractRetryer.java
+++ b/eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/AbstractRetryer.java
@@ -39,7 +39,7 @@ public abstract class AbstractRetryer {
         return retrys.size();
     }
 
-    protected void initDispatcher(Thread dispatcher) {
+    protected void initDispatcher() {
         dispatcher = new Thread(() -> {
             try {
                 DelayRetryable retryObj;

--- a/eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/grpc/retry/GrpcRetryer.java
+++ b/eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/grpc/retry/GrpcRetryer.java
@@ -56,7 +56,7 @@ public class GrpcRetryer extends AbstractRetryer {
             new EventMeshThreadFactory("grpc-retry", true, Thread.NORM_PRIORITY),
             new ThreadPoolExecutor.AbortPolicy());
 
-        initDispatcher(dispatcher);
+        initDispatcher();
     }
 
 }

--- a/eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/retry/HttpRetryer.java
+++ b/eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/retry/HttpRetryer.java
@@ -62,7 +62,7 @@ public class HttpRetryer extends AbstractRetryer {
             new EventMeshThreadFactory("http-retry", true, Thread.NORM_PRIORITY),
             new ThreadPoolExecutor.AbortPolicy());
 
-        initDispatcher(dispatcher);
+        initDispatcher();
     }
 
 }

--- a/eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/session/retry/EventMeshTcpRetryer.java
+++ b/eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/session/retry/EventMeshTcpRetryer.java
@@ -89,7 +89,7 @@ public class EventMeshTcpRetryer extends AbstractRetryer {
 
     @Override
     public void init() {
-        initDispatcher(dispatcher);
+        initDispatcher();
     }
 
     public void printRetryThreadPoolState() {

